src/java/org/apache/cassandra/config/Config.java

```diff
@@ -199,6 +200,8 @@ public MemtableOptions()
     // Limit the maximum depth of repair session merkle trees
     @Deprecated
     public volatile Integer repair_session_max_tree_depth = null;
+    @Mutable
+    @ValidatedBy(useClass = DATABASE_DESCRIPTOR_CLASS, useClassMethod = "validateRepairSessionSpace")
```

Comment:
@Mmuzaf I would expect that if a parameter is called "useClass", then the actual value of that parameter will be indeed a class.

    @Mutable
    @ValidatedBy(useClass = DatabaseDescriptor.class, useClassMethod = "validateRepairSessionSpace")

and the in the implementation you would use ".getName()" or something like that.

The minor problem with this approach is that if a class changes its name or location, this would not be updated.

I am not sure we can do the same for "method" as it is not so easy to reference here like class is but it is still an improvement.

Reply:
The problem with this approach is a circular dependency: the `DatabaseDescriptor` class has the `confValueAccessors` which are built from the `Config.class` and here we are trying to have a reference to the `DatabaseDescriptor.class` which is not good. So, since we still need to refer to a method by its name and considering the fact that the DatabaseDescriptor doesn't change its name or package at all because there are too many dependencies on it, we can refer to it by string representation. This seems to be a good compromise, wdyt?

Reply:
ok